rust/agama-network/src/nm/client.rs

```diff
@@ -411,6 +448,37 @@ impl<'a> NetworkManagerClient<'a> {
         Ok(proxy)
     }
 
+    async fn get_device_proxy(&self, name: String) -> Result<DeviceProxy, NmError> {
+        let proxy = NetworkManagerProxy::new(&self.connection).await?;
+        let mut device_path: Option<OwnedObjectPath> = None;
+        for path in &self.nm_proxy.get_all_devices().await? {
+            let proxy = DeviceProxy::builder(&self.connection)
+                .path(path.as_str())?
+                .build()
+                .await?;
+
+            if let Ok(device) = DeviceFromProxyBuilder::new(&self.connection, &proxy)
+                .build()
+                .await
+            {
+                if device.name == name {
```

Comment:
Can this happen more than one time? Because if that's not the case, you could break the loop here. You found what you were looking for.